x-pack/plugins/ml/server/models/bucket_span_estimator/polled_data_checker.js

```diff
@@ -13,111 +13,108 @@
 
 import { get } from 'lodash';
 
-export function polledDataCheckerFactory({ asCurrentUser }) {
-  class PolledDataChecker {
-    constructor(index, timeField, duration, query, runtimeMappings, indicesOptions) {
-      this.index = index;
-      this.timeField = timeField;
-      this.duration = duration;
-      this.query = query;
-      this.runtimeMappings = runtimeMappings;
-      this.indicesOptions = indicesOptions;
-
-      this.isPolled = false;
-      this.minimumBucketSpan = 0;
-    }
+export class PolledDataChecker {
```

Comment:
Removes `polledDataCheckerFactory` provider function and just exports the class.
`asCurrentUser` is now passed into the constructor.